qrenderdoc/Windows/Dialogs/TipsDialog.cpp

```diff
@@ -0,0 +1,303 @@
+/******************************************************************************
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2017 Baldur Karlsson
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#include "TipsDialog.h"
+#include "ui_TipsDialog.h"
+
+#include <cstdlib>
+
+TipsDialog::TipsDialog(QWidget *parent) : QDialog(parent), ui(new Ui::TipsDialog), m_currentTip(0)
+{
+  ui->setupUi(this);
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+  initialize();
+  showTip(m_currentTip);
+}
+
+TipsDialog::~TipsDialog()
+{
+  delete ui;
+}
+
+void TipsDialog::initialize()
+{
+  // Tip 1
+  m_tips.push_back(Tip{"Talk to me!",
```

Comment:
If you add a constructor with parameters then that hides the default constructor which is needed for putting the struct in a QVector, so you need to add it back again explicitly.